pkg/jsonnetfile/jsonnetfile.go

```diff
@@ -19,44 +19,84 @@ import (
 	"io/ioutil"
 	"os"
 
+	v0 "github.com/jsonnet-bundler/jsonnet-bundler/spec/v0"
+	v1 "github.com/jsonnet-bundler/jsonnet-bundler/spec/v1"
+	depsv1 "github.com/jsonnet-bundler/jsonnet-bundler/spec/v1/deps"
 	"github.com/pkg/errors"
-
-	"github.com/jsonnet-bundler/jsonnet-bundler/spec"
-	"github.com/jsonnet-bundler/jsonnet-bundler/spec/deps"
 )
 
 const (
 	File     = "jsonnetfile.json"
 	LockFile = "jsonnetfile.lock.json"
 )
 
-var ErrNoFile = errors.New("no jsonnetfile")
+var (
+	ErrNoFile   = errors.New("no jsonnetfile")
+	ErrUpdateJB = errors.New("jsonnetfile version unknown, update jb")
+)
 
 // Load reads a jsonnetfile.(lock).json from disk
-func Load(filepath string) (spec.JsonnetFile, error) {
+func Load(filepath string) (v1.JsonnetFile, error) {
 	bytes, err := ioutil.ReadFile(filepath)
 	if err != nil {
-		return spec.New(), err
+		return v1.New(), err
 	}
 
 	return Unmarshal(bytes)
 }
 
 // Unmarshal creates a spec.JsonnetFile from bytes. Empty bytes
 // will create an empty spec.
-func Unmarshal(bytes []byte) (spec.JsonnetFile, error) {
-	m := spec.New()
+func Unmarshal(bytes []byte) (v1.JsonnetFile, error) {
+	m := v1.New()
+
 	if len(bytes) == 0 {
 		return m, nil
 	}
-	if err := json.Unmarshal(bytes, &m); err != nil {
-		return m, errors.Wrap(err, "failed to unmarshal file")
+
+	versions := struct {
+		Version float64 `json:"version"`
+	}{}
+
+	err := json.Unmarshal(bytes, &versions)
+	if err != nil {
+		return m, err
 	}
-	if m.Dependencies == nil {
-		m.Dependencies = make(map[string]deps.Dependency)
+
+	if versions.Version > 1 {
+		return m, ErrUpdateJB
 	}
 
-	return m, nil
+	if versions.Version == 1 {
+		if err := json.Unmarshal(bytes, &m); err != nil {
+			return m, errors.Wrap(err, "failed to unmarshal v1 file")
+		}
+
+		return m, nil
+	} else {
+		var mv0 v0.JsonnetFile
+		if err := json.Unmarshal(bytes, &mv0); err != nil {
+			return m, errors.Wrap(err, "failed to unmarshal jsonnetfile")
+		}
+
+		for name, dep := range mv0.Dependencies {
```

Comment:
Converting `v0` to `v1` is pretty long here ... could you move that into a function to keep `Unmarshal` clean?

Reply:
I don't think that's too long to have it in here. This exactly the conversion from v0 to v1 that Unmarshal does here, that's the important part. Therefore I'd rather not hide it.